src/moveit_visual_tools.cpp

```diff
@@ -1527,20 +1527,13 @@ void MoveItVisualTools::publishRobotState(const moveit_msgs::DisplayRobotState&
 
 bool MoveItVisualTools::hideRobot()
 {
-  static const std::string VJOINT_NAME = "virtual_joint";
-
-  // Always load the robot state before using
-  loadSharedRobotState();
-
-  // Apply transform
-  Eigen::Isometry3d offset;
-  offset.translation().x() = rviz_visual_tools::LARGE_SCALE;
-  offset.translation().y() = rviz_visual_tools::LARGE_SCALE;
-  offset.translation().z() = rviz_visual_tools::LARGE_SCALE;
-  applyVirtualJointTransform(*hidden_robot_state_, offset);
+  moveit_msgs::DisplayRobotState display_robot_state_msg;
+  // Hide the robot state
+  display_robot_state_msg.hide = true;
 
   // Publish
-  return publishRobotState(hidden_robot_state_);
+  publishRobotState(display_robot_state_msg);
```

Comment:
Please, restore return value.

Reply:
Or should I restore these lines .? the code doesn't change hidden_robot_state_ anymore this's why I removed them

`loadSharedRobotState();`
`return publishRobotState(hidden_robot_state_);`

Reply:
I'm the original author, and I had a habit of all functions returning bool in case the use case came up of having an error state, which sometimes it didn't.

looking back, i think that was a bad design approach. however, regarding the scope of this PR, i think how @JafarAbdi has proposed is good.